File: Project phase Problem solving part/1200-minimum-absolute-difference/1200-minimum-absolute-difference.py
class Solution:
    def minimumAbsDifference(self, arr: List[int]) -> List[List[int]]:
        arr.sort()
        m = float('inf')
        
        for i in range(1, len(arr)):
            m = min(arr[i] - arr[i-1], m)
        
        res = []
        
        for i in range(1, len(arr)):
            if arr[i] - arr[i - 1] == m:
                res.append([arr[i - 1], arr[i]])
        
        return res

File: 1200-minimum-absolute-difference/1200-minimum-absolute-difference.js
/**
 * @param {number[]} arr
 * @return {number[][]}
 */
var minimumAbsDifference = function(arr) {
    arr.sort((a,b)=> a - b);
    const res = [];
    let min = Infinity;
    
    for(let i = 0; i < arr.length - 1; i++){
        let dif = arr[i + 1] - arr[i];
        min = Math.min(dif, min);
    }
    
    for(let i = 0; i < arr.length - 1; i++){
        let dif = arr[i + 1] - arr[i];
        if(min === dif){
            res.push([arr[i], arr[i + 1]]);
        }
    }
    return res
};